src/Lecture3.hs

```diff
@@ -246,4 +280,5 @@ Just [8,9,10]
 [8,20,3]
 
 -}
-apply = error "TODO"
+apply :: Functor f => a -> f (a -> a) -> f a
```

Comment:
You can generalize this function even further 🙂 There's no need to restrict the function to return values of the same type as an input argument. The implementation won't change at all!
```suggestion
apply :: Functor f => a -> f (a -> b) -> f b
```

Reply:
good point! thx!